File: app/src/main/java/com/yeonsik/fitnessapp/exercise/BodyPart.java
package com.yeonsik.fitnessapp.exercise;

public enum BodyPart {
    CHEST("chest", "가슴"),
    BACK("back", "등"),
    LEGS("legs", "하체"),
    SHOULDERS("shoulders", "어깨"),
    ARMS("arms", "팔"),
    ABS("abs", "복근");

    private final String id;
    private final String labelKo;

    BodyPart(String id, String labelKo) {
        this.id = id;
        this.labelKo = labelKo;
    }

    public String id() {
        return id;
    }

    public String labelKo() {
        return labelKo;
    }

    public static BodyPart fromId(String value) {
        if (value == null) {
            return null;
        }

        String normalized = value.trim().toLowerCase();
        for (BodyPart bodyPart : values()) {
            if (bodyPart.id.equals(normalized)) {
                return bodyPart;
            }
        }
        return null;
    }
}

File: shared/src/commonMain/kotlin/com/yeonsik/fitness/shared/feature/exercise/model/BodyPart.kt
package com.yeonsik.fitness.shared.feature.exercise.model

import kotlin.jvm.JvmStatic

enum class BodyPart(
    private val persistedId: String,
    private val koreanLabel: String
) {
    CHEST("chest", "가슴"),
    BACK("back", "등"),
    LEGS("legs", "하체"),
    SHOULDERS("shoulders", "어깨"),
    ARMS("arms", "팔"),
    ABS("abs", "복근");

    val id: String
        get() = persistedId

    fun id(): String = persistedId

    val labelKo: String
        get() = koreanLabel

    fun labelKo(): String = koreanLabel

    companion object {
        @JvmStatic
        fun fromId(value: String?): BodyPart? {
            if (value == null) return null
            val normalized = value.trim().lowercase()
            return values().firstOrNull { it.persistedId == normalized }
        }
    }
}
